src/app/code/community/Divante/VueStorefrontIndexer/Model/Resource/Catalog/Eav.php

```diff
@@ -183,15 +175,26 @@ private function getLoadAttributesSelect($storeId, $table, array $attributeIds,
                     'attribute_id',
                 ]
             )
-            ->joinLeft(
-                ['t_store' => $table],
-                $joinCondition,
-                ['value' => new Zend_Db_Expr('COALESCE(t_store.value, t_default.value)')]
-            )
             ->where('t_default.store_id = ?', 0)
             ->where('t_default.entity_id IN (?)', $entityIds)
             ->where('t_default.attribute_id IN (?)', $attributeIds);
 
+         if(($table == "catalog_product_entity_int" && $frontendInput != "system") ||
+           (($frontendInput == "multiselect" || ($frontendInput == "select") && $frontendInput != "system"))
+         ) {
+             $select->joinLeft(
+                ['v_store' => "eav_attribute_option_value"],
+                't_default.value=v_store.option_id',
+                ['value' =>  new Zend_Db_Expr('COALESCE(v_store.value, t_default.value)')]
+             );
+         } else {
+            $select->joinLeft(
+                ['t_store' => $table],
+                $joinCondition,
+                ['value' => new Zend_Db_Expr('COALESCE(t_store.value, t_default.value)')]
+            );
+         }
+
```

Comment:
I agree with Agata we should have an option_id, not the text value

Reply:
Well, for Elasticsearch aggregations and for displaying prorduct multiselect attributes in VSF, its necesary to have the attribute value instead of the integer option_id which is useless for the end-user. In the particular case of Magento EE, multiselect attributes are transmitted to ES as option_id instead of labels, which might not be the case for the CE. How would you like to proceed here ? An EE specific branch should be created for this ?

Reply:
multiselect attributes are transmitted to ES as option_id instead of labels, which might not be the case for the CE

I think it is the same for CE -> we have option ids in elasticearch. 
VSF (at least on version 1.7) is replacing option ids to labels on VSF Product Page. 

I see here two options:
1. We will add new option to configuration. So we can choose how we want to export select/multiselect attributes.
By default -> option ids
for multiselect/select default mapping should by int/long 
2. New branch for EE or separate module which will extend this one? 

The First option looks okay to me. 
@pkarw do you agree?

Reply:
@afirlejczyk yes, I do agree